### DISCUSSION TITLE
ExcelDnaUtil.Application

### DISCUSSION THREAD

**Discussion by hxhgxy** [Excel-DNA/ExcelDna#639] [Q&A] [answered]
![ExcelDNA](https://github.com/Excel-DNA/ExcelDna/assets/37101628/df7791db-fdb9-4985-abd0-be68febadc38)
I tried to get Excel Application in this way, refer to screen captured above please. My project is WPF Class Library with NET6.0

What is the right way? I will need to access workbook, worksheet, range etc later.

Thanks!

**Comment by govert** [comment]
The result from ExcelDnaUtil.Application is a Microsoft.Office.Interop.Excel.Application. Your code is trying to cast it to System.Windoes.Application.

**Comment by hxhgxy** [comment]
Do you have samples for NET6.0?

**Comment by govert** [comment, marked as answer]
Sure - this sample uses the COM object model to write to the sheet.
Note that it uses the ExcelDna.Interop package to get a reference to the Excel PIA assemblies.

**Reply by hxhgxy** [reply]
Sorry, but where can I download the sample?
